### PR TITLE
New package: Ananicy

### PR DESCRIPTION
**Pull request by linuxergr**
Closes issue https://github.com/void-linux/void-packages/issues/26429

#### General
- [X] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [X] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [X] I built this PR locally for my native architecture, (ARCH-LIBC)

### CODE REVIEW

File: srcpkgs/ananicy/template [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'ananicy'
+pkgname=ananicy
+version=2.2.1
+revision=1
+wrksrc=Ananicy-${version}
+depends="bash schedtool python3"
```
Comment (paper42) [inline srcpkgs/ananicy/template:6]:
```suggestion
depends="schedtool python3"
```

File: srcpkgs/ananicy/template [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+distfiles="https://github.com/Nefelim4ag/Ananicy/archive/refs/tags/${version}.tar.gz"
+checksum=3e33c3c84d29f93b64079e67eb56c39a4d40b0591e52db171156981b72f648b1
+
+do_install() {
+	make install PREFIX="${DESTDIR}"
+}
+
+post_install() {
+	rm -rf "${DESTDIR}/lib"
+	chmod +x "${DESTDIR}/usr/bin/ananicy"
```
Comment (paper42) [inline srcpkgs/ananicy/template:20]: Why is this necessary? Doesn't the makefile make this executable.

File: srcpkgs/ananicy/template [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+short_desc="Ananicy daemon"
+maintainer="linuxer <linuxer@disroot.org>"
+license="GPL-3.0-only"
+homepage="https://github.com/Nefelim4ag/Ananicy"
+distfiles="https://github.com/Nefelim4ag/Ananicy/archive/refs/tags/${version}.tar.gz"
+checksum=3e33c3c84d29f93b64079e67eb56c39a4d40b0591e52db171156981b72f648b1
+
+do_install() {
+	make install PREFIX="${DESTDIR}"
+}
```
Comment (paper42) [inline srcpkgs/ananicy/template:16]: It would be nice to upstream a fix to support DESTDIR.

File: srcpkgs/ananicy/files/ananicy/run [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+printf '0 \n' > /proc/sys/kernel/sched_autogroup_enabled
```
Comment (paper42) [inline srcpkgs/ananicy/files/ananicy/run:3]: I think we should leave this to the user and document it.

File: srcpkgs/ananicy/template [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'ananicy'
+pkgname=ananicy
+version=2.2.1
+revision=1
+wrksrc=Ananicy-${version}
+depends="bash schedtool python3"
+short_desc="Ananicy daemon"
```
Comment (Duncaen) [inline srcpkgs/ananicy/template:7]: The description is useless. Maybe:
> Another auto nice daemon, with community rules support

File: srcpkgs/ananicy/files/ananicy/log/run [hunk 6]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+exec logger -t ananicy
```
Comment (Duncaen) [inline srcpkgs/ananicy/files/ananicy/log/run:2]: This should use `vlogger(1)` instead of logger.

File: srcpkgs/ananicy/template (hunk 1 above)
Comment (linuxergr) [reply srcpkgs/ananicy/template:6]: Well, not all use bash and base-minimal does not include it, so adding it enforces it's normal operation under any conditions.

File: srcpkgs/ananicy/template (hunk 2 above)
Comment (linuxergr) [reply srcpkgs/ananicy/template:20]: yup, this can be removed

File: srcpkgs/ananicy/files/ananicy/run (hunk 4 above)
Comment (mobinmob) [reply srcpkgs/ananicy/files/ananicy/run:3]: I think that is a bad idea, because it leads to a service that does not really work ootb. The script currently follows what the [upstream systemd service](https://github.com/Nefelim4ag/Ananicy/blob/master/ananicy.service) does, but uses printf instead of sysctl as the later is not exactly suited for use in chainloading...

I cooperated with @linuxer in order to create and test an [ananicy service](https://github.com/mobinmob/void-66-services/blob/devel/usr/share/66/service/ananicy) for void-66-services. In an execline script sysctl is used in an if block. In a posix shell script, printf seems like a good choice.

File: srcpkgs/ananicy/files/ananicy/log/run (hunk 6 above)
Comment (linuxergr) [reply srcpkgs/ananicy/files/ananicy/log/run:2]: Was based on another package as an example on how to use the logger, so I used the same one:

https://github.com/void-linux/void-packages/blob/master/srcpkgs/earlyoom/files/earlyoom/log/run

File: srcpkgs/ananicy/template (hunk 3 above)
Comment (linuxergr) [reply srcpkgs/ananicy/template:16]: Thanks for your suggestion ;)

File: srcpkgs/ananicy/files/ananicy/log/run (hunk 6 above)
Comment (linuxergr) [reply srcpkgs/ananicy/files/ananicy/log/run:2]: Done, btw :)

File: srcpkgs/ananicy/template (hunk 1 above)
Comment (paper42) [reply srcpkgs/ananicy/template:6]: But where does it use bash? It's a python script that doesn't call bash.

File: srcpkgs/ananicy/files/ananicy/log/run (hunk 6 above)
Comment (paper42) [reply srcpkgs/ananicy/files/ananicy/log/run:2]: this was not fixed

File: srcpkgs/earlyoom/files/earlyoom/log/run [hunk 7]
```diff
@@ -1,2 +1,2 @@
 #!/bin/sh
-exec logger -t earlyoom
+exec vlogger -t earlyoom
```
Comment (paper42) [inline srcpkgs/earlyoom/files/earlyoom/log/run:2]: this is an unrelated change

**Review by paper42 — COMMENTED** [review]
make sure this PR contains only one commit

File: srcpkgs/earlyoom/files/earlyoom/log/run (hunk 7 above)
Comment (linuxergr) [reply srcpkgs/earlyoom/files/earlyoom/log/run:2]: yup, wonder how was mixed :/
